admin/controllers/school.js

```diff
@@ -444,14 +444,47 @@ const getSelectPupilNotTakingCheck = async (req, res, next) => {
 
   return res.render('school/select-pupils-not-taking-check', {
     breadcrumbs: req.breadcrumbs(),
+    sortField,
+    sortDirection,
     attendanceCodes,
-    formData,
     pupilsList,
     htmlSortDirection,
     arrowSortDirection
   })
 }
 
+const savePupilNotTakingCheck = async (req, res, next) => {
+  res.locals.pageTitle = 'Save pupils not taking the check'
+  req.breadcrumbs(res.locals.pageTitle)
+
+  if (req.body.attendanceCode === undefined || req.body.pupil === undefined) {
+    return res.redirect('/school/pupils-not-taking-check/select-pupils')
+  }
+
+  const todayDate = moment(moment.now()).format()
+  let pupils = req.body.pupil
+
+  pupils.forEach(async (id) => {
+    let pupil = await fetchOnePupil(id, req.user.School)
```

Comment:
This is fetching each pupil in sequence, and likely to be slower than needed.  Why not fetch them all in parallel?

Reply:
Done.